From 6f38842a9955d179c816707f8d28835e842d6278 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Thu, 6 Jul 2017 16:12:12 -0500
Subject: [PATCH 6/6] SensitivePhoneNumbers: Fix number comparison

* Currently, we just compare the strings for equality,
  which results in incorrect detection of sensitive nums
  a lot of the time, because adding (or removing) the
  country code is enough to make the detection fail,
  meaning the call to that phone number is logged
* Use Android's PhoneNumberUtils comparison method to
  fix this, since it takes these factors into account

Change-Id: I26ac180f8a6552cf87a4bada1d370f0ebb884ee1
---
 .../android/server/telecom/SensitivePhoneNumbers.java  | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/com/android/server/telecom/SensitivePhoneNumbers.java b/src/com/android/server/telecom/SensitivePhoneNumbers.java
index b5703b2c..eff6d6ee 100644
--- a/src/com/android/server/telecom/SensitivePhoneNumbers.java
+++ b/src/com/android/server/telecom/SensitivePhoneNumbers.java
@@ -19,6 +19,7 @@ package com.android.server.telecom;
 
 import android.content.Context;
 import android.os.Environment;
+import android.telephony.PhoneNumberUtils;
 import android.telephony.SubscriptionManager;
 import android.telephony.TelephonyManager;
 import android.text.TextUtils;
@@ -111,8 +112,13 @@ public class SensitivePhoneNumbers {
         String networkUsed = telephonyManager.getNetworkOperator(subIdInt);
         if (!TextUtils.isEmpty(networkUsed)) {
             String networkMCC = networkUsed.substring(0, 3);
-            return mSensitiveNumbersMap.containsKey(networkMCC) &&
-                   mSensitiveNumbersMap.get(networkMCC).contains(numberToCheck);
+            if (mSensitiveNumbersMap.containsKey(networkMCC)) {
+                for (String num : mSensitiveNumbersMap.get(networkMCC)) {
+                    if (PhoneNumberUtils.compare(numberToCheck, num)) {
+                        return true;
+                    }
+                }
+            }
         }
         return false;
     }
-- 
2.17.1

